### ISSUE TITLE
training speed has slowed down since the recent updates

### ISSUE THREAD

**Issue by miraiaroha** [TideDra/lmm-r1#41]
Hello!
We find that the training speed has slowed down after the recent updates.
Few weeks before about 143s/it, now 233s/it.
My shell is:
`GPU_NUM=4
ray job submit --address="http://127.0.0.1:8265" \
   --runtime-env-json='{"working_dir": "/data0/user"}' \
   -- python3 -m openrlhf.cli.train_ppo_ray \
   --ref_num_nodes 1 \
   --ref_num_gpus_per_node 4 \
   --remote_rm_url ./examples/scripts/reward_func.py \
   --actor_num_nodes 1 \
   --actor_num_gpus_per_node 4 \
   --vllm_num_engines 4 \
   --vllm_tensor_parallel_size 1 \
   --colocate_all_models \
   --vllm_enable_sleep \
   --vllm_gpu_memory_utilization 0.55 \
   --vllm_sync_backend gloo \
   --enable_prefix_caching \
   --pretrain $PRETRAIN_MODEL \
   --save_path $SAVE_PATH/$MODEL_CPK_NAME \
   --micro_train_batch_size 1 \
   --train_batch_size 4 \
   --micro_rollout_batch_size 2 \
   --rollout_batch_size 8 \
   --temperature 0.9 \
   --n_samples_per_prompt 6 \
   --max_epochs 1 \
   --num_episodes 16 \
   --prompt_max_len 10000 \
   --max_samples 100000 \
   --generate_max_len 2000 \
   --advantage_estimator rloo \
   --zero_stage 3 \
   --bf16 \
   --actor_learning_rate 1e-6 \
   --init_kl_coef 0.001 \
   --prompt_data $DATASET \
   --input_key message \
   --label_key answer \
   --normalize_reward \
   --flash_attn \
   --gradient_checkpointing \
   --save_steps 10000 \
   --ckpt_path ${SAVE_PATH}/${MODEL_CPK_NAME}/ckpt \
   --save_hf_ckpt \
   --use_tensorboard ${SAVE_PATH}/${MODEL_CPK_NAME}/logs`

**Comment by TideDra** [comment]
Can you measure the speed of each stage (sampling, making experience, training)? The new version is synced with openrlhf, it may be blamed to upstream updates.

**Comment by miraiaroha** [comment]
> Can you measure the speed of each stage (sampling, making experience, training)? The new version is synced with openrlhf, it may be blamed to upstream updates.

Oh, it's my mistake. I comapred qwen2-vl-2b with qwen2.5-vl-3b. That is normal that the latter is 50% slower than the former.

By the way, why the latest code ONLY support qwen2.5, qwen2 is not supported.

**Comment by miraiaroha** [comment]
> Since we only conduct our experiments on qwen2.5-vl now, which is stronger and more suitable for r1-reproduction, we didn't transer qwen2vl into the new version. However, it's easy to support qwen2vl in the new version, since it shares most code with qwen2.5vl and our codebase is designed for easy extensibility. To support qwen2vl, you basically just need to copy the folder of qwen2_5_vl, rename all 2.5 to 2, and apply the modifications between 2 and 2.5 to the hacked functions.

Thinks!
